File: cores/mobile-flutter/lib/src/core/upload/file_category.dart
/// Upload file categories aligned with `@enistere/api-contracts`.
enum FileCategory {
  image,
  document,
  avatar,
  media,
  video,
  audio,
  identityDocument,
  attachment,
  other,
}

extension FileCategoryExtension on FileCategory {
  String get apiValue => switch (this) {
    FileCategory.image => 'IMAGE',
    FileCategory.document => 'DOCUMENT',
    FileCategory.avatar => 'AVATAR',
    FileCategory.media => 'MEDIA',
    FileCategory.video => 'VIDEO',
    FileCategory.audio => 'AUDIO',
    FileCategory.identityDocument => 'IDENTITY_DOCUMENT',
    FileCategory.attachment => 'ATTACHMENT',
    FileCategory.other => 'OTHER',
  };
}

File: cores/web-angular/src/app/core/upload/file-category.ts
export type FileCategory =
  | 'IMAGE'
  | 'DOCUMENT'
  | 'AVATAR'
  | 'VIDEO'
  | 'AUDIO'
  | 'IDENTITY_DOCUMENT'
  | 'ATTACHMENT'
  | 'MEDIA'
  | 'OTHER';

export const FILE_CATEGORIES: { value: FileCategory; label: string }[] = [
  { value: 'DOCUMENT', label: 'Document' },
  { value: 'IMAGE', label: 'Image' },
  { value: 'AVATAR', label: 'Avatar' },
  { value: 'VIDEO', label: 'Vidéo' },
  { value: 'AUDIO', label: 'Audio' },
  { value: 'IDENTITY_DOCUMENT', label: "Pièce d'identité" },
  { value: 'ATTACHMENT', label: 'Pièce jointe' },
  { value: 'MEDIA', label: 'Média' },
  { value: 'OTHER', label: 'Autre' },
];
